Implement scripts and compatibility check

Gianluca005 opened the pull request:
Add license compatibility check

Gianluca005 committed:
Implement static layer checks and enhance project cleanup script; add Makefile for build automation

Gianluca005 committed:
Remove unused format_code.py script

Gianluca005 committed:
Refactor import paths and enhance LicenseCompatibilityAnalyzer initialization; update PackageMetadataFetcher to return dependency graph

kbublij2 commented on src/analyzer/main.py:
```diff
@@ -8,26 +10,49 @@
 
 logger = LoggerFormatter.initialize(__name__, logging.DEBUG)
 
+PROJECT_ROOT = Path(__file__).resolve().parents[2]
```
Altered path resolution slightly and added folder creation notice to the docstring.

kamilok04 committed:
Apply autogenerated fixes

Fixes:
Missing/incomplete docstrings
Translation issues
Incorrect path resolutions
Missing type annotations

kbublij2 reviewed (commented):
More refactoring will happen shortly

kamilok04 committed:
Implement basic refactoring

Package metadata caching
Detect dependency cycles
Improve the license mapping function
Begin creating unit tests
Don't attempt repo downloads when no URL is found

Known problems:
- Some license names remain unnormalized when
  comparing
- Unclear behaviour after failure to determine
  license

kamilok04 committed:
Fix unnormalized license names

kamilok04 committed:
Refactor license mapper into a separate file

kamilok04 committed:
Improved I/O handling

Create cache mechanisms to avoid excessive I/O
Create cache overrides to do I/O anyway if needed
Fix path resolution and file handling

Note: project is intended to be run from its
root directory.

kamilok04 committed:
Implement first round of pylint fixes

Implement missing doctrings
Break up complex logic into separate methods
Fix various typos and incorrect EOLs

kamilok04 committed:
Apply second round of Pylint fixes

Refactor I/O methods
Remove (for now) unneeded files
Outline complex logic into debuggable methods

kamilok04 committed:
Reimplement the license comparator

Establish scancode result caching
Generate tree comparison reports
Implement the Strategy pattern for lic. comparator
Handle potential multiple scanned licenses
Handle unknown licenses gracefully
Pylint 10/10

kamilok04 committed:
Merge branch 'develop' into implement-scripts

kamilok04 committed:
Preparation for merging

Fixed faulty path recognition
Filled missing docstrings
Resolved merge conflicts

kamilok04 committed:
Preparation for merging

Fixed faulty path recognition
Filled missing docstrings
Resolved merge conflicts

kamilok04 committed:
Merge branch 'implement-scripts' of https://github.com/License-Compliance-Group/LicenseSentinel into implement-scripts

kbublij2 reviewed (approved):
All the pending changes have been succesfully implemented
